tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/temporal/TableauEncoder.scala

```diff
@@ -0,0 +1,444 @@
+package at.forsyte.apalache.tla.pp.temporal
+
+import at.forsyte.apalache.tla.lir._
+
+import javax.inject.Singleton
+import com.typesafe.scalalogging.LazyLogging
+import scalaz.Scalaz.{init => _}
+import at.forsyte.apalache.tla.pp.UniqueNameGenerator
+import at.forsyte.apalache.tla.pp.IrrecoverablePreprocessingError
+import at.forsyte.apalache.tla.pp.temporal.utils.builder
+import at.forsyte.apalache.tla.pp.temporal.ScopedBuilderExtensions._
+import at.forsyte.apalache.tla.pp.temporal.DeclUtils._
+import scala.collection.mutable.HashMap
+import at.forsyte.apalache.tla.lir.oper.TlaTempOper
+import at.forsyte.apalache.tla.typecomp.TBuilderInstruction
+import at.forsyte.apalache.tla.lir.transformations.TransformationTracker
+
+/**
+ * Encodes temporal formulas as invariants.
+ *
+ * @author
+ *   Philip Offtermatt
+ */
+@Singleton
+class TableauEncoder(
+    module: TlaModule,
+    gen: UniqueNameGenerator,
+    loopEnc: LoopEncoder,
+    tracker: TransformationTracker)
+    extends LazyLogging {
+  val levelFinder = new TlaLevelFinder(module)
+  val varNamesToExStrings = new HashMap[String, String]()
+
+  def encodeVarNameMapping(modWithPreds: ModWithPreds): ModWithPreds = {
```

Comment:
This is essentially a filler method, it will be refactored by the more informative counterexample writer, I don't think it's worth documenting since it will be gone in the next PR.

For context, it generated the table you saw in the temporal encoding a while ago, e.g.
```
GenVarNamesToExFun ==
  { { "loop_t_9", "☐(∀i$10 ∈ Corr: (pc[i$10] ≠ 'AC'))" },
    { "loop_t_8",
      "(∀i$9 ∈ Corr: (pc[i$9] = 'V0')) ⇒ ☐(∀i$10 ∈ Corr: (pc[i$10] ≠ 'AC'))" },
    { "t_8",
      "(∀i$9 ∈ Corr: (pc[i$9] = 'V0')) ⇒ ☐(∀i$10 ∈ Corr: (pc[i$10] ≠ 'AC'))" },
    { "t_9", "☐(∀i$10 ∈ Corr: (pc[i$10] ≠ 'AC'))" } }
```